Python/pytime.c

```diff
@@ -262,6 +262,20 @@ _PyTime_FromNanoseconds(long long ns)
     return t;
 }
 
+int
+_PyTime_FromMicroseconds(_PyTime_t *tp, long long us)
+{
+    Py_BUILD_ASSERT(sizeof(long long) <= sizeof(_PyTime_t));
+    _PyTime_t t = (_PyTime_t)us;
+
+    if (_PyTime_check_mul_overflow(t, US_TO_NS)) {
+        _PyTime_overflow();
```

Comment:
I don't think it will be safe to raise an exception from a PyThread function (which could be called without the GIL).

Reply:
Oops, you're right. I forgot that.